ingestion-beam/src/main/java/com/mozilla/telemetry/contextualservices/SponsoredInteraction.java

```diff
@@ -0,0 +1,108 @@
+package com.mozilla.telemetry.contextualservices;
+
+import com.google.auto.value.AutoValue;
+import org.apache.beam.sdk.coders.*;
+import org.apache.beam.sdk.io.gcp.pubsub.PubsubMessage;
+import org.apache.beam.sdk.io.gcp.pubsub.PubsubMessageWithAttributesCoder;
+import org.apache.beam.sdk.values.TypeDescriptor;
+
+import javax.annotation.Nullable;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.io.Serializable;
+import java.util.Map;
+
+@AutoValue
+public abstract class SponsoredInteraction implements Serializable {
+    // topsites, suggest
+    public static final String SOURCE_TOPSITE = "topsites";
+    public static final String SOURCE_SUGGEST = "suggest";
+    abstract String source();
+
+    // desktop, phone, tablet
+    public static final String FORM_DESKTOP = "desktop";
+    public static final String FORM_PHONE = "phone";
+    public static final String FORM_TABLET = "tablet";
+    abstract String form();
+
+    // impression, click
+    public static final String INTERACTION_IMPRESSION = "impression";
+    public static final String INTERACTION_CLICK = "click";
+    abstract String interaction();
+
+    abstract String contextID();
```

Comment:
@jklukas can context_id be nullable?

Reply:
It's [marked as required in the desktop schemas](https://github.com/mozilla-services/mozilla-pipeline-schemas/blob/2f5f9d54926eee33b7de0889d8f2c853fed948bf/schemas/contextual-services/topsites-impression/topsites-impression.1.schema.json#L74) but can't be guaranteed at the schema level for the Glean case. I would expect it to always be present.